### DISCUSSION TITLE
Upgrade to v22.0.5 causes type errors because of t returning null

### DISCUSSION THREAD

**Discussion by medihack** [i18next/i18next#1866] [Q&A] [answered]
There are several locations in my React code where I use the `t` function to provide a translation to a label prop of a 3rd party component library. Unfortunately, those label props only accept `string` or `undefined` type. Everything was ok with i18next v22.0.4, but v22.0.5 broke the typing as the TFunction now returns `null` instead of `undefined` (error message below). I think I understand the rationale behind this change (fix type errors when used in react components), but what is the best way to handle `t` function in this scenario? Just do something like `t("foobar") ?? undefined` or `t("foobar")!`?

```
Type 'DefaultTFuncReturn' is not assignable to type 'string | undefined'.
        Type 'null' is not assignable to type 'string | undefined'.
```

**Comment by pedrodurek** [comment, marked as answer]
`t` function can return `null`, this behaviour is set by default, if you want to change it, set `returnNull` type to `false`.
```ts
// i18next.d.ts
import 'i18next';

declare module 'i18next' {
  interface CustomTypeOptions {
    returnNull: false;
    ...
  }
}
```

I also recommend updating your `i18next` configuration to behave accordantly. https://www.i18next.com/overview/configuration-options#translation-defaults
```ts
i18next.init({
  returnNull: false,
   ...
});
```

**Reply by medihack** [reply]
@pedrodurek Thanks for your help. Unfortunately, I have some Typescript problems after creating the `i18next.d.ts` file (in the root folder of my project) with the above content (of course, without the `...`). Some imported types (e.g. `import { Resource } from "i18next"`) can't be resolved anymore (`Module '"i18next"' has no exported member 'Resource'`). I also included the definition file in my `tsconfig.json`, but this doesn't solve the problem. Do you have an idea how to fix this?

**Reply by pedrodurek** [reply]
Hey @medihack, try to place the type definition file under the `src` path, like in this example: https://github.com/i18next/i18next/blob/master/examples/typescript/%40types/i18next.d.ts

**Reply by medihack** [reply]
Ah, ok. I get it now! Thanks a lot for the help again. I mark it as answered.
